communitter-frontend/src/components/InvitationModal.js

```diff
@@ -0,0 +1,114 @@
+import React, { useState } from "react";
+import { Button, Modal, Table } from "react-bootstrap";
+import { BsTrash } from "react-icons/bs";
+import { setErrorMessage } from "../features/errorSlice";
+import { url } from "../utilities/config";
+import { fetchWithOpts } from "../utilities/fetchWithOptions";
+import NewInvitationModal from "./NewInvitationModal";
+import { useDispatch } from "react-redux";
+
+const InvitationModal = ({ show, onHide }) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const dispatch = useDispatch();
+
+  const [showNewInvitationModal, setShowNewInvitationModal] = useState(false);
+
+  const handleOpenNewInvitationModal = () => setShowNewInvitationModal(true);
+  const handleCloseNewInvitationModal = () => setShowNewInvitationModal(false);
+
+  const handleAddInvitation = async (newInvitation) => {
+    setIsLoading(true);
+    try {
+      await fetchWithOpts(`${url}/invitation`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(newInvitation),
+      });
+      setShowNewInvitationModal(false);
+      setIsLoading(false);
+      console.log(newInvitation);
+    } catch (err) {
+      dispatch(setErrorMessage(err.message));
+      setIsLoading(false);
+    }
+  };
+
+  // Example invitation data
+  const invitationData = [
+    {
+      communityName: "Sample Community",
+      invitedBy: "John Doe",
+      username: "johndoe123",
+      role: "Member",
+      date: "2024-10-28",
+    },
+  ];
+
+  const handleCancelInvitation = (userId) => {
+    alert(`Invitation for ${userId} has been canceled.`);
+  };
+
+  return (
+    <>
+      <Modal show={show} onHide={onHide} size="lg" centered>
+        <Modal.Header closeButton>
+          <Modal.Title>Invitation Details</Modal.Title>
+          <Button
+            variant="primary"
+            onClick={handleOpenNewInvitationModal}
+            className="ms-3"
+          >
+            New Invitation
+          </Button>
+        </Modal.Header>
+        <Modal.Body>
+          <Table bordered responsive>
+            <thead>
+              <tr>
+                <th>Username</th>
+                <th>Role</th>
+                <th>Invited By</th>
+                <th>Date</th>
+                <th>Actions</th>
+              </tr>
+            </thead>
+            <tbody>
+              {invitationData.map((invitation, index) => (
```

Comment:
we map through a mock data, where did you define the real data, or are you goind to define it?